src/coreclr/jit/layout.h

```diff
@@ -35,6 +35,9 @@ class ClassLayout
         BYTE  m_gcPtrsArray[sizeof(BYTE*)];
     };
 
+    // The normalized type to use in IR for block nodes with this layout.
+    const var_types m_type;
```

Comment:
This increases the size of `ClassLayout` by one pointer. I measured the overall impact on memory consumption and CG-ing x64 CoreLib, it looked insignificant:
```
./diff-mem x64
Relative difference is: 0.007%
```
(This is a memory-CPU tradeoff: `impNormStructType` is not a cheap function)